components/Themes/index.js

```diff
@@ -40,50 +40,71 @@ const ThemeItem = ({ children, item, isSelected, onClick }) => (
 
 const themeIcon = <ThemeIcon />
 
+const getCustomName = themes =>
+  `Custom Theme ${themes.filter(({ name }) => name.startsWith('Custom Theme')).length + 1}`
+
 class Themes extends React.PureComponent {
   selectedTheme = DEFAULT_THEME
 
   state = {
     themes: THEMES,
     preset: this.props.theme,
-    highlights: {},
     name: 'Custom Theme',
     selected: null
   }
 
   dropdown = React.createRef()
 
   componentDidMount() {
+    const { update, theme, highlights } = this.props
     const storedThemes = getThemes(localStorage) || []
 
     this.setState(({ themes }) => {
       const newThemes = [...storedThemes, ...themes]
 
-      const name = `Custom Theme ${newThemes.filter(({ name }) => name.startsWith('Custom Theme'))
-        .length + 1}`
+      const name = getCustomName(newThemes)
+
+      this.selectedTheme = newThemes.find(({ id }) => id === theme) || DEFAULT_THEME
 
-      this.selectedTheme = newThemes.find(({ id }) => id === this.props.theme) || DEFAULT_THEME
+      if (Object.keys(highlights).length === 0) {
+        update({ highlights: this.selectedTheme.highlights })
+      }
 
       return {
         themes: newThemes,
-        highlights: this.selectedTheme.highlights,
         name
       }
     })
   }
 
-  applyPreset = preset =>
-    this.setState(({ themes }) => ({
-      preset,
-      highlights: themes.find(({ id }) => id === preset).highlights
-    }))
+  componentDidUpdate(prevProps) {
+    const { isVisible, theme, update } = this.props
+    const { themes } = this.state
+
+    if (prevProps.isVisible && !isVisible) {
+      this.setState({ name: getCustomName(themes) })
```

Comment:
Why does `name` have to be in state? Can it not be derived?

Reply:
This sets the initial state for the name which will then be updated as you type into the input.

Reply:
Hopefully 72a77c8 makes that more clear?

Reply:
I don't care either way 🙂feel free to keep this or revert. If you want it to be `input`, I'd prefer it be called input everwhere and not renamed to `name` in _some_ places.

Reply:
I don't care either way 🙂feel free to keep this or revert. If you want it to be `input`, I'd prefer it be called input everwhere and not renamed to `name` in _some_ places.